src/main/webapp/WEB-INF/view/searchbar.jsp

```diff
@@ -1,45 +1,53 @@
-<%@ page import="codeu.model.data.User" %>
-<%@ page import="codeu.model.store.basic.UserStore" %>
-<%@ page import="java.util.List" %>
-<%
-List<User> allUsers = UserStore.getInstance().getAllUsers();
-%>
-
 <script type="text/javascript">
-  // Checks if the search request is empty.
-  function isEmpty() {
-    var searchBar = document.getElementById("searchBar");
-    if(searchBar.value.length < 1) {
-      alert("Please enter a valid search.");
-      return false;
-    }
-    return true;
-  }
-
-  // Toggle drop down.
-  function toggleDropDown() {
-    var searchBar = document.getElementById("searchBar");
-    var dropDiv = document.getElementById("dropDiv");
-    if (searchBar.value == "") {
-      dropDiv.style.display = "none";
+  // Iterates through fetched results
+  function fetchResults() {
+    var searchBarValue = document.querySelector('#searchBar').value;
+    if (searchBarValue.length > 0) {
+      fetch('/search?searchRequest=' + searchBarValue, {credentials: "same-origin"})
+        .then(
+          function(response) {
+            if (response.status !== 200) {
+              console.log("Looks like there was a problem. Status Code: " + response.status);
+              return;
+            }
+            response.json().then(function(data) {
+              if (data != null) {
+                // Start with no resultItem divs
+                document.querySelector('#result').innerHTML = '';
+                for (var user in data) {
+                  var userName = data[user].name;
+                  var div = document.createElement("div");
+                  div.setAttribute("class", "resultItem");
+                  a = document.createElement("a");
+                  a.href = "/profile/" + userName;
+                  a.innerHTML = userName;
+                  div.appendChild(a);
+                  document.getElementById("result").appendChild(div);
+                }
+              }
+              // Data is null
+              else {
+                // Clear the resultItem divs
+                document.querySelector('#result').innerHTML = '';
+              }
+            });
+          }
+        ) 
+        .catch(function(err) {
```

Comment:
nit: you could make use of the more compact lambda syntax. Basically, these two are (mostly*) equivalent:
```
catch(function(err) {
  console.log(...);
});
```
and:
```
catch(err => console.log(...));
```
----
\* there are some semantic differences.